assets/src/components/form/base/with-form-handler.js

```diff
@@ -66,30 +66,32 @@ export const withFormHandler = ( FormComponent ) => {
 		/**
 		 * @function
 		 */
-		async componentDidMount() {
+		componentDidMount() {
 			if ( isFunction( this.props.loadHandler ) ) {
 				this.setState( { loading: true } );
-				const data = await this.props.loadHandler();
+				const data = this.props.loadHandler();
 				this.setState( { loading: false, data } );
 			}
 		}
 
 		render() {
 			const {
 				loading,
-				submitHandler,
 				resetHandler,
 				errorMessage = '',
 				loadingNotice = '',
 				decorators = [],
 				mutators = {},
 				formChanges = false,
+				setMutatorCallbacks = () => null,
 				...formProps
 			} = this.props;
 			let { formData = null } = this.props;
 			formData = formData === null && ! isEmpty( this.state.data ) ?
 				this.state.data :
 				formData;
+			const showSubmit = typeof this.props.submitHandler === 'function';
```

Comment:
hmm I don't know if its faster or not.  I just sometimes switch to typeof statements :)  I think generally speaking we should:

- use lodash functions if there's already need for something in the file.
- avoid lodash functions if there's one or two places where vanilla javascript is sufficient.

In this case I probably could have used `isFunction` here because `isEmpty` is already being used.